tests/e2e/nvidia-cdi-refresh_test.go

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"text/template"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"sigs.k8s.io/kind/pkg/build/nodeimage"
+)
+
+const (
+	nvidiaCdiRefreshInstallTemplate = `
+	# Read the TMPDIR
+	TMPDIR=$(cat /tmp/ctk_e2e_temp_dir.txt)
+	export TMPDIR
+
+	# uninstall the nvidia-container-toolkit
+	apt-get remove -y nvidia-container-toolkit nvidia-container-toolkit-base libnvidia-container-tools libnvidia-container1
+	apt-get autoremove -y
+
+	# Remove the cdi file if it exists
+	if [ -f /var/run/cdi/nvidia.yaml ]; then
+		rm -f /var/run/cdi/nvidia.yaml
+	fi
+
+	# Stop the nvidia-cdi-refresh.path and nvidia-cdi-refresh.service units
+	systemctl stop nvidia-cdi-refresh.path
+	systemctl stop nvidia-cdi-refresh.service
+	
+	# Reload the systemd daemon
+	systemctl daemon-reload
+	
+	# Start the dummy service to force systemd to enter a degraded state
+	cat <<EOF > /etc/systemd/system/dummy.service
+[Unit]
+Description=Dummy systemd service
+
+[Service]
+Type=oneshot
+ExecStart=/usr/bin/sh -c "exit 0"
+EOF
+
+	# We know the dummy service will fail, so we can ignore the error
+	systemctl start dummy.service 2>/dev/null || true
+
+	# Install the nvidia-container-toolkit
+	dpkg -i ${TMPDIR}/packages/ubuntu18.04/amd64/libnvidia-container1_*_amd64.deb ${TMPDIR}/packages/ubuntu18.04/amd64/nvidia-container-toolkit-base_*_amd64.deb ${TMPDIR}/packages/ubuntu18.04/amd64/libnvidia-container-tools_*_amd64.deb
+
+	if ! systemctl status nvidia-cdi-refresh.path | grep "Active: active"; then
+		echo "nvidia-cdi-refresh.path is not Active"
+		exit 1
+	fi
```

Comment:
Should this be an `It` clause?

Reply:
Agree, I am now moving this to a stand alone template, and calling it from it's own `It` block